platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -27,37 +29,47 @@ @implementation NSExpression (MGLAdditions)
         return { std::string([(NSString *)value UTF8String]) };
     } else if ([value isKindOfClass:NSNumber.class]) {
         NSNumber *number = (NSNumber *)value;
-        // handle all types defined in NSNumber designated initializers
-        if (strcmp([number objCType], @encode(char)) == 0) {
-            return { number.charValue };
-        } else if (strcmp([number objCType], @encode(unsigned char)) == 0) {
-            return { number.unsignedCharValue };
-        } else if (strcmp([number objCType], @encode(short)) == 0) {
-            return { number.shortValue };
-        } else if (strcmp([number objCType], @encode(unsigned short)) == 0) {
-            return { number.unsignedShortValue };
-        } else if (strcmp([number objCType], @encode(int)) == 0) {
-            return { number.intValue };
-        } else if (strcmp([number objCType], @encode(unsigned int)) == 0) {
-            return { number.unsignedIntValue };
-        } else if (strcmp([number objCType], @encode(long)) == 0) {
-            return { number.longValue };
-        } else if (strcmp([number objCType], @encode(unsigned long)) == 0) {
-            return { number.unsignedLongValue };
-        } else if (strcmp([number objCType], @encode(long long)) == 0) {
-            return { number.longLongValue };
-        } else if (strcmp([number objCType], @encode(unsigned long long)) == 0) {
-            return { number.unsignedLongLongValue };
-        } else if (strcmp([number objCType], @encode(float)) == 0) {
-            return { number.floatValue };
+        NSLog(@"NSNumber %@ (%s)", number, number.objCType);
+
+        // 32/64-bit check based on CGFloat storage size
+        // per https://developer.apple.com/library/ios/documentation/General/Conceptual/CocoaTouch64BitGuide/
+        BOOL thirtyTwoBit = sizeof(CGFloat) == sizeof(float);
```

Comment:
I found a number of other ways that work, but this definitely does, reads cleanly, and has a concrete documentation link explaining the approach. So maybe just keep it?